machine/machine_core/machine_virtual.py

```diff
@@ -305,7 +305,18 @@ def __init__(
 
         Machine.__init__(self, image=image, **kwargs)
 
-        self.run_dir = os.path.join(os.getenv("TEST_OVERLAY_DIR", "/var/tmp"), "bots-run")
+        overlay_dir = os.getenv("TEST_OVERLAY_DIR")
+        if not overlay_dir:
+            # toolbox compatibility: /tmp is shared with the host, but may be too small for big overlays (tmpfs!)
+            # $HOME is shared, but we don't want to put our junk there (NFS, backups)
+            # /var/tmp is not shared with the host but the right place; just in case session libvirtd is already
+            # running, use the shared path so that the daemon can actually see our overlay.
+            # But this only makes sense if the host also has /run/host set up (toolbox ships a tmpfiles.d)
+            if os.path.exists("/run/host/var/tmp") and os.path.exists("/run/host/run/host"):
```

Comment:
Maybe it is not an issue but I see these paths even on the host system. I guess it's just fine.

Reply:
Yes, that's intentional. If you have toolbox installed, you should have a `/run/host → ../` symlink (and thus you can do infinite recursion :grin: ). Without toolbox installed, /run/host doesn't exist, but then you also can't run cockpit in toolbox.

Reply:
Hmm, interesting. I wonder if the same applies with `distrobox`?